### PR TITLE
chore: [NCCL] reorg and better error messages

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Precommit was clearly not run, assorted fixes and reorg. Some AIs for @apbose re: custom ops and some hardcoded special cases 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/select.py [hunk 1]
```diff
@@ -25,6 +23,8 @@
 )
 from torch_tensorrt.fx.types import TRTTensor
 
+import tensorrt as trt
+
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/conversion/impl/select.py:27]: Why is this change here? Is this a pre-commit change?

File: py/torch_tensorrt/dynamo/lowering/passes/fuse_distributed_ops.py [hunk 2]
```diff
@@ -12,16 +12,23 @@
 logger = logging.getLogger(__name__)
 
 
-def tensorrt_fused_nccl_all_gather_op(args0, args1, args2):
+# TODO: @apbose make these actual torch custom ops, should allow aot_export
+def tensorrt_fused_nccl_all_gather_op(
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/lowering/passes/fuse_distributed_ops.py:16]: Ok noted in the TODOs. Just to get clarity, what is meant by allowing aot_export? What is the difference in the present behavior?

File: pyproject.toml [hunk 3]
```diff
@@ -101,17 +102,12 @@ include-package-data = false
 
 [tool.uv]
 environments = ["sys_platform == 'linux'", "sys_platform == 'windows'"]
-
 prerelease = "if-necessary-or-explicit"
-
+index-strategy = "unsafe-best-match"                                    # Needed for TRT-LLM
```
Comment (apbose) [inline pyproject.toml:107]: Why is this added? What is the significance?

**Comment by apbose** [comment]
Wanted to get clarity on-` precommit was clearly not run`. I generally run pre-commit run, and push in the modified changes and cross check if the CI Lint check passes. Should I be mindful of other things before pushing in?
Also what are the hardcoded changes?
Have some review comments/questions, else the PR changes look good to me.

**Review by apbose — APPROVED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -85,9 +85,10 @@ dev = [
 torchvision = [
     "torchvision",
 ] #Leaving torchvisions dependency unconstrained so uv can just install something that should work for the torch we have. TV's on PyT makes it hard to put version constrains in
-quantization = ["nvidia-modelopt[deploy,hf,torch]~=0.17.0"]
+quantization = ["nvidia-modelopt[deploy,hf,torch]>=0.17.0"]
 monitoring-tools = ["rich>=13.7.1"]
 jupyter = ["rich[jupyter]>=13.7.1"]
+distributed = ["tensorrt-llm>=0.16.0"]
```
Comment (apbose) [inline pyproject.toml:92]: How do we ensure this goes through when we have a torchTRT container having python version different than 3.10. Won't this cause an issue here?

Comment (narendasan) [reply pyproject.toml:92]: Its an extra so it wont effect install

File: py/torch_tensorrt/dynamo/conversion/impl/select.py (hunk 1 above)
Comment (narendasan) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:27]: Precommit

File: py/torch_tensorrt/dynamo/lowering/passes/fuse_distributed_ops.py (hunk 2 above)
Comment (narendasan) [reply py/torch_tensorrt/dynamo/lowering/passes/fuse_distributed_ops.py:16]: Sonthat you dont need to use the autograd workflow and could potentially support torch.export flows

File: pyproject.toml (hunk 3 above)
Comment (narendasan) [reply pyproject.toml:107]: This is for uv to install Tensorrt llm

**Comment by narendasan** [comment]
> Wanted to get clarity on-` precommit was clearly not run`. I generally run pre-commit run, and push in the modified changes and cross check if the CI Lint check passes. Should I be mindful of other things before pushing in? Also what are the hardcoded changes? Have some review comments/questions, else the PR changes look good to me.

There were a number of issues regarding type annotations in code added in the PR, so at some point it was pushed without verifying through precommit

**Comment by apbose** [comment]
Ok when I run lint check on the previous MR added files I see the below
lint_check
I thought linting takes care of this and if the CI Lint check passes, then these should be good.

Looks like these need to be taken care of manually then?
